tests/unit/apps/test_base.py

```diff
@@ -41,3 +43,93 @@ def test_application_get_latest_os_version_failed(
         app._get_latest_os_version(unit)
 
     mock_find_compatible_versions.assert_called_once_with(charm, unit.workload_version)
+
+
+@pytest.mark.parametrize("config", ({}, {"enable-auto-restarts": {"value": True}}))
+@patch("cou.apps.base.OpenStackApplication.channel", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication._verify_channel")
+def test_check_auto_restarts(_, _channel, config):
+    """Test function to verify that enable-auto-restarts is disabled."""
+    app_name = "app"
+    app = OpenStackApplication(app_name, MagicMock(), config, MagicMock(), "", "ch", "")
+
+    app._check_auto_restarts()
+
+
+@patch("cou.apps.base.OpenStackApplication.channel", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication._verify_channel")
+def test_check_auto_restarts_error(*_):
+    """Test function to verify that enable-auto-restarts is disabled raising error."""
+    app_name = "app"
+    exp_error_msg = (
+        "COU does not currently support upgrading applications that disable service restarts. "
+        f"Please enable charm option enable-auto-restart and rerun COU to upgrade the {app_name} "
+        "application."
+    )
+    config = {"enable-auto-restarts": {"value": False}}
+    app = OpenStackApplication(app_name, MagicMock(), config, MagicMock(), "", "ch", "")
+
+    with pytest.raises(ApplicationError, match=exp_error_msg):
+        app._check_auto_restarts()
+
+
+@patch("cou.apps.base.OpenStackApplication.channel", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication._verify_channel")
+@patch("cou.apps.base.OpenStackApplication.apt_source_codename", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication.current_os_release", new_callable=PropertyMock)
+@patch(
+    "cou.apps.base.OpenStackApplication.can_upgrade_current_channel",
+    new_callable=PropertyMock(return_value=False),
+)
+def test_check_application_target(
+    can_upgrade_current_channel, current_os_release, apt_source_codename, *_
+):
+    """Test function to verify target."""
+    target = OpenStackRelease("victoria")
+    release = OpenStackRelease("ussuri")
+    app_name = "app"
+    app = OpenStackApplication(app_name, MagicMock(), {}, MagicMock(), "", "ch", "")
+    current_os_release.return_value = apt_source_codename.return_value = release
+
+    app._check_application_target(target)
+
+
+@patch("cou.apps.base.OpenStackApplication.channel", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication._verify_channel")
+@patch("cou.apps.base.OpenStackApplication.apt_source_codename", new_callable=PropertyMock)
+@patch("cou.apps.base.OpenStackApplication.current_os_release", new_callable=PropertyMock)
+@patch(
+    "cou.apps.base.OpenStackApplication.can_upgrade_current_channel",
+    new_callable=PropertyMock(return_value=False),
+)
```

Comment:
non-blocking: maybe its better to include a test case with `can_upgrade_current_channel.return_value=True` (and other conditions as `false`)

Reply:
You are right, but I will add it in another PR, since this one is blocking me.